src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -584,11 +585,47 @@ impl<'a> TypeChecker<'a> {
                 if !is_builtin_function(func_name)
                     && !Self::all_rewritable_scalar_function().contains(&func_name)
                 {
-                    return self.resolve_udf(*span, func_name, args).await;
+                    if let Some(udf) = self.resolve_udf(*span, func_name, args).await? {
+                        return Ok(udf);
+                    } else {
+                        // Function not found, try to find and suggest similar function name.
+                        let all_funcs = BUILTIN_FUNCTIONS
+                            .all_function_names()
+                            .into_iter()
+                            .chain(AggregateFunctionFactory::instance().registered_names())
+                            .chain(
+                                Self::all_rewritable_scalar_function()
+                                    .iter()
+                                    .cloned()
+                                    .map(str::to_string),
+                            );
+                        let mut engine: SimSearch<String> = SimSearch::new();
+                        for func_name in all_funcs {
+                            engine.insert(func_name.clone(), &func_name);
+                        }
+                        let possible_funcs = engine
```

Comment:
```
mysql> select ai_txo_sql('this funtion is ok?');
ERROR 1105 (HY000): Code: 1008, displayText = error:
  --> SQL:1:8
  |
1 | select ai_txo_sql('this funtion is ok?')
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no function matches the given name: ai_txo_sql
```
Looks like we need this `suggest` crate:
https://crates.io/crates/suggest

Reply:
`ai_to_sql` is a table function thus we can only use it like `select * from ai_to_sql()`?

Reply:
Oh sorry, I see.